src/Compilers/CSharp/Portable/Symbols/Source/SourceOrdinaryMethodSymbol.cs

```diff
@@ -632,7 +632,24 @@ internal sealed override bool CallsAreOmitted(SyntaxTree syntaxTree)
             return base.CallsAreOmitted(syntaxTree);
         }
 
-        internal sealed override bool GenerateDebugInfo => !IsAsync && !IsIterator;
```

Comment:
This needed to change because the initial implementation method previously got replaced with a compiler-generated stub body, and we would never generate debug info. We now look at whether runtime async is enabled for this, but as I haven't added dedicated testing I am leaving a prototype to follow up.